enterprise/models.py

```diff
@@ -4692,13 +4692,17 @@ def _get_filtered_ecu_ids(self, user_query):
         # https://docs.djangoproject.com/en/5.0/topics/security/
         var_q = f"%{user_query}%"
         sql_string = """
-            select entcu.id from enterprise_enterprisecustomeruser entcu
-            join auth_user au on entcu.user_id = au.id
-            where entcu.enterprise_customer_id = %s and au.email like %s;
+            with users as (
+                select ecu.id, au.email, coalesce(aup.name, concat(au.first_name, ' ', au.last_name)) as full_name
+                from enterprise_enterprisecustomeruser ecu
+                inner join auth_user au on ecu.user_id = au.id
+                left join auth_userprofile aup on au.id = aup.user_id
+                where ecu.enterprise_customer_id = %s
+            ) select * from users where email like %s or full_name like %s;
```

Comment:
in the previous query we were only selecting the `id` column from `enterprise_enterprisecustomeruser`, should we still be doing that? (instead of `select *`)